expression/builtin_compare.go

```diff
@@ -407,12 +407,14 @@ func resolveType4Extremum(args []Expression) types.EvalType {
 	if aggType.EvalType().IsStringKind() {
 		for i := range args {
 			item := args[i].GetType()
-			if types.IsTemporalWithDate(item.Tp) {
-				temporalItem = item
+			if types.IsTemporal(item.Tp) {
```

Comment:
Please add some comments about why do we need to do it.

Reply:
OK.